File: app/src/main/java/com/clpstudio/bsocial/core/persistance/ISharedPreferencesUtils.java
package com.clpstudio.bsocial.core.persistance;

import android.content.SharedPreferences;

/**
 * Created by DealDash
 * Interface for interacting with the shared preferences in androud
 */
public interface ISharedPreferencesUtils {

    /**
     * Register listener for shared preferences changes
     * @param listener The listener
     */
    void registerOnChangeListener(SharedPreferences.OnSharedPreferenceChangeListener listener);

    /**
     * Unregister listener for shared preferences changes
     * @param listener The listener to be unregistered
     */
    void unregisterOnChangeListener(SharedPreferences.OnSharedPreferenceChangeListener listener);



}

File: app/src/main/java/com/clpstudio/bsocial/core/persistance/ISharedPreferencesUtils.kt
package com.clpstudio.bsocial.core.persistance

import android.content.SharedPreferences

/**
 * Interface for interacting with the shared preferences in androud
 */
interface ISharedPreferencesUtils {

    /**
     * Register listener for shared preferences changes
     * @param listener The listener
     */
    fun registerOnChangeListener(listener: SharedPreferences.OnSharedPreferenceChangeListener)

    /**
     * Unregister listener for shared preferences changes
     * @param listener The listener to be unregistered
     */
    fun unregisterOnChangeListener(listener: SharedPreferences.OnSharedPreferenceChangeListener)


}
